docs/src/main/asciidoc/ap4k.adoc

```diff
@@ -0,0 +1,155 @@
+For the time being it is recreated to manually create said `Dockerfile` (something will change soon with the proper `Dockerfile` being provided when the project is created using the Maven `create` goal), by creating `src/main/docker/Dockerfile.jvm` with the following content:
+
+[source, subs=attributes+]
+----
+FROM fabric8/java-jboss-openjdk8-jdk:1.5.4
+ENV JAVA_OPTIONS=-Dquarkus.http.host=0.0.0.0
+COPY target/lib/* /deployments/lib/
+COPY target/*-runner.jar /deployments/app.jar
+ENTRYPOINT [ "/deployments/run-java.sh" ]
+----
+
+Now, add the `io.ap4k.docker.annotation.EnableDockerBuild` annotation to `MyApplication` like so:
+
+[source,java]
+----
+import javax.ws.rs.core.Application;
+import io.ap4k.docker.annotation.EnableDockerBuild;
+import io.ap4k.kubernetes.annotation.KubernetesApplication;
+import io.ap4k.kubernetes.annotation.Port;
+
+@EnableDockerBuild(dockerFile = "src/main/docker/Dockerfile.jvm")
+@KubernetesApplication(ports = @Port(name = "http", containerPort = 8080), group = "yourDockerUsername")
+public class MyApplication extends Application {
+
+}
+----
+
+If you start out by simply executing `mvn clean package -Dap4k.build=true` then at the end of the build you have a `yourDockerUsername/ap4k:1.0-SNAPSHOT` image built locally that can be executed like so:
+
+`docker run --rm -p 8080:8080 yourDockerUsername/ap4k:1.0-SNAPSHOT`
+
+To actually deploy your application onto your Kubernetes cluster, you need to execute `mvn clean package -Dap4k.push=true -Dap4k.deploy=true`.
+
+In addition to what the previous command did, this command also pushes the image to the registry and applies the Kubernetes resources. At the end of the process the application will be deployed onto the cluster.
+
+== Enable Openshift support
+
+For this section of the guide it is assumed that `oc` is present on the PATH and that is able to properly communicate with the Openshift cluster you intend to use.
+
+Deploying applications onto Openshift is even easier than vanilla Kubernetes due to the existence of https://docs.openshift.com/container-platform/3.6/architecture/core_concepts/builds_and_image_streams.html#source-build[S2I].
+
+To enable Openshift and S2I support, modify `MyApplication` as so:
+
+[source,java]
+----
+import javax.ws.rs.core.Application;
+import io.ap4k.kubernetes.annotation.Env;
+import io.ap4k.kubernetes.annotation.Port;
+import io.ap4k.openshift.annotation.EnableS2iBuild;
+import io.ap4k.openshift.annotation.OpenshiftApplication;
+
+@OpenshiftApplication(envVars = @Env(name = "JAVA_OPTIONS", value = "-Dquarkus.http.host=0.0.0.0"), ports = @Port(name = "http", containerPort = 8080))
+@EnableS2iBuild
+public class MyApplication extends Application {
+
+}
+----
+
+If you simply execute `mvn clean package` you will see two new files named `openshift.json` and `openshift.yaml`
+in `target/wiring-classes/META-INF/ap4k/` directory. These contain the Openshift specific resources.
```

Comment:
Same comment about the target directory maybe not being the one I would have used.

Reply:
Same response as above :)